src/local_scheduler/local_scheduler_algorithm.cc

```diff
@@ -1090,10 +1100,9 @@ void handle_worker_blocked(LocalSchedulerState *state,
       /* Add the worker to the list of blocked workers. */
       worker->is_blocked = true;
       utarray_push_back(algorithm_state->blocked_workers, &worker);
-      /* Return the resources that the blocked worker was using. */
-      CHECK(worker->task_in_progress != NULL);
-      TaskSpec *spec = Task_task_spec(worker->task_in_progress);
-      update_dynamic_resources(state, spec, true);
+      /* Return the resources that the blocked worker was using, but not GPU
+       * resources because it will still be using memory on those GPUs. */
+      release_resources(state, worker, (double) worker->cpus_in_use, 0);
```

Comment:
For the sake of symmetry, it might be nicer to call `release_resources` with the same arguments as the corresponding `acquire_resources` call:
```
release_resources(state, worker,
                  TaskSpec_get_required_resource(spec, ResourceIndex_CPU),
                  0);
```